packages/engine/src/avatar/components/TransparencyDitheringComponent.tsx

```diff
@@ -0,0 +1,177 @@
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { Entity, defineComponent, getOptionalComponent, useComponent, useEntityContext } from '@etherealengine/ecs'
+import { defineState, getMutableState, matches, useHookstate } from '@etherealengine/hyperflux'
+import { matchesVector3 } from '@etherealengine/spatial/src/common/functions/MatchesUtils'
+import { addOBCPlugin } from '@etherealengine/spatial/src/common/functions/OnBeforeCompilePlugin'
+import { RendererState } from '@etherealengine/spatial/src/renderer/RendererState'
+import { MeshComponent } from '@etherealengine/spatial/src/renderer/components/MeshComponent'
+import { isArray } from 'lodash'
+import React, { useEffect } from 'react'
+import { FrontSide, Material, Vector3 } from 'three'
+import { SceneComponent } from '../../scene/components/SceneComponent'
+import { useModelSceneID } from '../../scene/functions/loaders/ModelFunctions'
+import { MaterialLibraryState } from '../../scene/materials/MaterialLibrary'
+import {
+  ditheringAlphatestChunk,
+  ditheringFragUniform,
+  ditheringVertex,
+  ditheringVertexUniform
+} from '../functions/ditherShaderChunk'
+
+export enum ditherCalculationType {
+  worldTransformed = 1,
+  localPosition = 0
+}
+
+export const transparencyDitheringState = defineState({
+  name: 'transparencyDitheringState',
+  initial: { materialIds: {} as Record<Entity, string[]> }
+})
+
+export const maxDitherPoints = 4 //should be equal to the length of the vec3 array in the shader
+export const TransparencyDitheringComponent = Array.from({ length: maxDitherPoints }, (_, i) => {
+  return defineComponent({
+    name: `TransparencyDitheringComponent${i}`,
+    onInit: (entity) => {
+      return {
+        center: new Vector3(),
+        distance: 2,
+        exponent: 2,
+        calculationType: ditherCalculationType.worldTransformed,
+        //internal
+        materialIds: [] as string[]
+      }
+    },
+
+    onSet: (entity, component, json) => {
+      if (!json) return
+      if (matchesVector3.test(json.center)) component.center.set(new Vector3().copy(json.center))
+      if (matches.number.test(json.distance)) component.distance.set(json.distance)
+      if (matches.number.test(json.exponent)) component.exponent.set(json.exponent)
+      if (matches.number.test(json.calculationType)) component.calculationType.set(json.calculationType)
+    },
+
+    reactor: () => {
```

Comment:
we should use a query reactor here instead of a new reactor for each component in the array